File: dart/04-control-flow/conditionals.dart
void main() {
  int temperature = 18;

  // Classic if / else if / else
  if (temperature > 30) {
    print('It is hot');
  } else if (temperature >= 15) {
    print('It is mild');
  } else {
    print('It is cold');
  }

  // Ternary conditional expression: condition ? valueIfTrue : valueIfFalse
  String label = temperature >= 15 ? 'comfortable' : 'chilly';
  print('Today is $label');

  // Null-aware ?? operator: use the right side when the left is null
  String? nickname;
  print('Hello, ${nickname ?? 'guest'}');
}

File: swift/04-control-flow/conditionals.swift
let temperature = 18

// if / else if / else
if temperature <= 0 {
    print("Freezing")
} else if temperature < 15 {
    print("Cold")
} else if temperature < 25 {
    print("Mild")
} else {
    print("Hot")
}

// The ternary conditional operator: condition ? valueIfTrue : valueIfFalse
let score = 72
let result = score >= 60 ? "Pass" : "Fail"
print("Result: \(result)")

// Conditions must be Bool — Swift has no "truthy" integers
let isReady = true
if isReady {
    print("Ready to go")
}
